### DISCUSSION TITLE
how to distinguish a variable from static,local,global with token

### DISCUSSION THREAD

**Discussion by cloud-yu** [microsoft/vscode-cpptools#8208] [Q&A] [answered]
```
int g_gid;
static int g_frame;

void fun1()
{
    int temp;
    static int cnt;
...
}
```
global variable `g_gid` and `g_frame`  colored by token "variable.global"
`temp` and `cnt` colored by token "variable.static"
Specifically I want to change the color of  variables modifierd by "static", is this possible?

**Comment by Colengms** [comment, marked as answer]
Hi @cloud-yu .  The C/C++ extension will emit semantic tokens for `local` and `global` variables, but it don't believe it currently identifies `static` variables.  I would suggest [opening a feature request (as an issue)](https://github.com/microsoft/vscode-cpptools/issues/new?assignees=&labels=&template=feature-request.md).

Customizing the colorization of `local` and `global` could be done using a setting, like so:

```
    "editor.semanticTokenColorCustomizations": {
        "rules": {
            "*.global": "#ff0000",
            "*.local": "#ff00ff"
        }
    },
```
